### PR TITLE
Feature/convert to poetry

### PR DESCRIPTION
**Pull request by SamRWest**
Converted from setup.py to pyproject.toml based poetry build for a reproducible environment. 
Dropped python3.7 support as the dependency tree can't be solved (e.g. networkx requires >=3.8)

### CODE REVIEW

File: tests/test_ditto_cli.py
Comment (SamRWest) [inline tests/test_ditto_cli.py:80]: Added this function instead of just subprocess.Popen so that stderr outputs get printed instead of swallowed

File: .github/workflows/ci.yml
Comment (SamRWest) [inline .github/workflows/ci.yml:18]: Some of the libraries (e.g. networkx & subdependencies) no longer support 3.7, so I've removed it and updated to 3.10 here

File: tests/test_ditto_cli.py
Comment (SamRWest) [inline tests/test_ditto_cli.py:21]: `poetry run` is now necessary (on windows only) for these commands to work (from within the unit tests only)

File: setup.py.old
Comment (SamRWest) [inline setup.py.old:1]: Disabled the old setup.py for now.  pyproject should be a direct replacement for it.  Will delete in future once it's all working.

File: .github/workflows/publish-to-pypi.yml
Comment (SamRWest) [inline .github/workflows/publish-to-pypi.yml:8]: Only (attempt to) publish to pypi after a merge to master now, not on every push.

File: .github/workflows/ci.yml
Comment (Bis-Bala) [reply .github/workflows/ci.yml:18]: We may need to keep up pace with the latest python version.

File: tests/test_ditto_cli.py
Comment (Bis-Bala) [reply tests/test_ditto_cli.py:21]: subprocess Popen is really flexible and also we can control over the processes to work with. Just to be aware that it can make slower when many files are open or does all sort of checking in cross-platform support.

File: tests/test_ditto_cli.py
Comment (Bis-Bala) [reply tests/test_ditto_cli.py:80]: Cool feature shlex - lexical analysis of shell style syntaxes.  Very much useful when we deal particularly with nested quotes.

**Review by Bis-Bala — APPROVED** [review]
Thanks Sam. Looks good to me.

File: .github/workflows/ci.yml
Comment (SamRWest) [reply .github/workflows/ci.yml:18]: Yeah good point, maybe 3.11 in future.

File: tests/test_ditto_cli.py
Comment (SamRWest) [reply tests/test_ditto_cli.py:21]: The subprocess call is just a quick (existing) unit test to make sure the ditto-cli entrypoint works, so no performance issues here.

File: tests/test_ditto_cli.py
Comment (SamRWest) [reply tests/test_ditto_cli.py:80]: Yeah, that was new to me too.  Added to the toolbox :)

**Comment by SamRWest** [comment]
Thanks Bis.
(ps. normally you'd have to hit the approve button after the review for the PR creator to be able to merge in github)

**Pull request merged**
